addons/mail/static/src/new/core/messaging.js

```diff
@@ -114,7 +113,7 @@ export class Messaging {
             channelMembers: {},
             /** @type {Object.<number, import("@mail/new/core/notification_model").Notification>} */
             notifications: {},
-            notificationGroups: {},
+            notificationGroups: [],
```

Comment:
I'm not fond of just relying on array for ordering and storing.
Should keep `{}` for pure storing, and dedicated `[]` for ordering